File: python/L5_constructor_overloading_workaround.py
class Student:
    def __init__(self, name="Unknown", age=0):
        self.name = name
        self.age = age

    def display(self):
        print(f"Name: {self.name}, Age: {self.age}")

s1 = Student() # Uses default arguments
s1.display()

s2 = Student("Lohit") # Overrides the default name
s2.display()

s3 = Student("Lohit P T", 21) # Provides all arguments
s3.display()

File: javascript/codes/L5_constructor_overloading_workaround.js
class Student {
    constructor(name = "Unknown", age = 0) {
        this.name = name;
        this.age = age;
    }

    display() {
        console.log(`Name: ${this.name}, Age: ${this.age}`);
    }
}

const s1 = new Student(); // Uses default parameters
s1.display();

const s2 = new Student("Lohit"); // Overrides the default name
s2.display();

const s3 = new Student("Lohit P T", 21); // Provides all arguments
s3.display();
